File: app/src/main/java/com/martinlaizg/geofind/data/access/database/converter/PlayLevelTypeConverter.java
package com.martinlaizg.geofind.data.access.database.converter;

import com.martinlaizg.geofind.data.enums.PlayLevel;

import androidx.room.TypeConverter;

public class PlayLevelTypeConverter {


	@TypeConverter
	public static PlayLevel toPlayLevel(String playLevelString) {
		return playLevelString == null ?
				null :
				playLevelString.isEmpty() ?
						PlayLevel.MAP :
						PlayLevel.valueOf(playLevelString);
	}

	@TypeConverter
	public static String fromPlayLevel(PlayLevel playLevel) {
		return playLevel == null ?
				"" :
				playLevel.toString();
	}
}

File: app/src/main/java/com/martinlaizg/geofind/data/access/database/converter/PlayLevelTypeConverter.kt
package com.martinlaizg.geofind.data.access.database.converter

import androidx.room.TypeConverter
import com.martinlaizg.geofind.data.enums.PlayLevel

object PlayLevelTypeConverter {
    @kotlin.jvm.JvmStatic
    @TypeConverter
    fun toPlayLevel(playLevelString: String?): PlayLevel? {
        return if (playLevelString == null) null else if (playLevelString.isEmpty()) PlayLevel.MAP else PlayLevel.valueOf(playLevelString)
    }

    @kotlin.jvm.JvmStatic
    @TypeConverter
    fun fromPlayLevel(playLevel: PlayLevel?): String {
        return playLevel?.toString() ?: ""
    }
}